src/Mod/PartDesign/Gui/TaskExtrudeParameters.cpp

```diff
@@ -1211,6 +1224,119 @@ void TaskExtrudeParameters::handleLineFaceNameNo()
+            lengthGizmo1->getDraggerContainer()->visible = true;
+            lengthGizmo2->getDraggerContainer()->visible = false;
+            taperAngleGizmo1->getDraggerContainer()->visible = true;
+            taperAngleGizmo2->getDraggerContainer()->visible = false;
+            break;
+        case Mode::TwoDimensions:
+            lengthGizmo1->getDraggerContainer()->visible = true;
+            lengthGizmo2->getDraggerContainer()->visible = true;
+            taperAngleGizmo1->getDraggerContainer()->visible = true;
+            taperAngleGizmo2->getDraggerContainer()->visible = true;
+            break;
+        default:
+            lengthGizmo1->getDraggerContainer()->visible = false;
+            lengthGizmo2->getDraggerContainer()->visible = false;
+            taperAngleGizmo1->getDraggerContainer()->visible = false;
+            taperAngleGizmo2->getDraggerContainer()->visible = false;
+        }
+    });
+
+    connect(ui->checkBoxReversed, &QCheckBox::toggled, [taperAngleGizmo1, lengthGizmo1, taperAngleGizmo2, lengthGizmo2](bool) {
+        GizmoPlacement placement = lengthGizmo1->getDraggerPlacement();
+        SoLinearDraggerContainer* draggerContainer = lengthGizmo1->getDraggerContainer();
+        lengthGizmo1->setDraggerPlacement(placement.pos, -draggerContainer->getPointerDirection());
+        // This is to ensure that the rotation dragger is positioned correctly after the reverse
+        draggerContainer->getDragger()->translation.touch();
+
+        placement = lengthGizmo2->getDraggerPlacement();
+        draggerContainer = lengthGizmo2->getDraggerContainer();
+        lengthGizmo2->setDraggerPlacement(placement.pos, -draggerContainer->getPointerDirection());
+        draggerContainer->getDragger()->translation.touch();
+
+        placement = taperAngleGizmo1->getDraggerPlacement();
+        taperAngleGizmo1->setDraggerPlacement(placement.pos, -taperAngleGizmo1->getDraggerContainer()->getPointerDirection());
+
+        placement = taperAngleGizmo2->getDraggerPlacement();
+        taperAngleGizmo2->setDraggerPlacement(placement.pos, -taperAngleGizmo2->getDraggerContainer()->getPointerDirection());
+    });
+
+    gizmos->addGizmo(lengthGizmo1);
+    gizmos->addGizmo(lengthGizmo2);
+    gizmos->addGizmo(taperAngleGizmo1);
+    gizmos->addGizmo(taperAngleGizmo2);
+    gizmos->initGizmos();
+
+    setGizmoPositions();
+
+    vp->attachGizmos(gizmos.get());
+
+    // These gizmos shouldn't be visible until the user set type to two dimensions
+    taperAngleGizmo2->getDraggerContainer()->visible = false;
+    lengthGizmo2->getDraggerContainer()->visible = false;
+}
+
+void TaskExtrudeParameters::setGizmoPositions()
+{
+    if (!gizmos) {
+        return;
+    }
+
+    auto lengthGizmo1 = gizmos->getGizmo<LinearGizmo>(0);
+    auto lengthGizmo2 = gizmos->getGizmo<LinearGizmo>(1);
+    auto taperAngleGizmo1 = gizmos->getGizmo<RotationGizmo>(2);
+    auto taperAngleGizmo2 = gizmos->getGizmo<RotationGizmo>(3);

```

Comment:
Yup, that reinforces my feeling that you should not create and store `Gizmos` instance in Task panels. Instead create field for each Gizmo and store them. Using indices to obtain objects is a big code smell and we must avoid it.

Reply:
Should be fixed.